com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -39,23 +41,136 @@
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitEnvironmentDivision(EnvironmentDivisionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitDataDivision(DataDivisionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitDataDivisionSection(DataDivisionSectionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitEnvironmentDivisionBody(EnvironmentDivisionBodyContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitProcedureSectionHeader(ProcedureSectionHeaderContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
   }
 
   @Override
   public Class visitProcedureSection(ProcedureSectionContext ctx) {
-    String wrongToken = ctx.getStart().getText();
-    throwWarning(wrongToken, ctx.getStart().getLine(), ctx.getStart().getCharPositionInLine());
+    Token token = ctx.getStart();
+    throwWarning(token.getText(), token.getLine(), token.getCharPositionInLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitParagraph(ParagraphContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitDataDescriptionEntryFormat1(DataDescriptionEntryFormat1Context ctx) {
+    Token token = ctx.getStart();
+    String tokenText = token.getText();
+    if (tokenText.equals("01") || tokenText.equals("1") || tokenText.equals("77")) {
+      areaAWarning(ctx.start.getCharPositionInLine(), tokenText, token.getLine());
+    }
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitProcedureDeclaratives(ProcedureDeclarativesContext ctx) {
+    Token firstDeclarative = ctx.getStart();
+    int firstDeclLine = firstDeclarative.getLine();
+    Token declarativeBody = ctx.procedureDeclarative(0).getStart();
+    Token endToken = ctx.END().getSymbol();
+
+    if (firstDeclLine == declarativeBody.getLine()) {
+      throwException(
+          declarativeBody.getText(),
+          declarativeBody.getLine(),
+          declarativeBody.getCharPositionInLine(),
+          "Following token can not be on the same line with DECLARATIVE token: ",
```

Comment:
Is better provide a ```String``` constant for quick reference

Reply:
sure